src/hpa.c

```diff
@@ -541,9 +576,7 @@ hpa_alloc_psset(tsdn_t *tsdn, hpa_shard_t *shard, size_t size) {
 		malloc_mutex_unlock(tsdn, &shard->grow_mtx);
 		return NULL;
 	}
-	if (hpa_should_hugify(shard, ps)) {
-		hpdata_hugify_allowed_set(ps, true);
-	}
+	hpa_update_purge_hugify_eligibility(shard, ps);
```

Comment:
Should this be included in the _update_end below, so that we never forget it?

Reply:
I think not; this is the hpa/psset split from above.